src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -434,14 +434,24 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state, GCHand
                 // Set maximum automatic redirection option
                 SetCurlOption(requestHandle, CURLoption.CURLOPT_MAXREDIRS, _maxAutomaticRedirections);
             }
-            if (state.RequestMessage.Content != null)
+
+            if (state.RequestMessage.Method == HttpMethod.Put)
```

Comment:
The previous code used Content != null here.  Is that expected for this case?  Should we assert it?